### PR TITLE
KEB OIDC default values

### PR DESCRIPTION
**Pull request by szwedm**
**Description**

Due to recent update to provisioning parameters with OIDC values, KEB requires default OIDC values in case these are not provided by user.

Changes proposed in this pull request:

- add configuration for default OIDC values.

**Related issue(s)**
See also #779

### CODE REVIEW

**Review by PK85 — APPROVED** [review]
I'm approving API design and flows.

File: components/kyma-environment-broker/internal/process/input/input.go [hunk 1]
```diff
@@ -376,8 +377,15 @@ func (r *RuntimeInput) adjustRuntimeName() error {
...
-		// TODO: read and use default values
+	if !r.provisioningParameters.Parameters.OIDC.IsProvided() {
+		r.provisionRuntimeInput.ClusterConfig.GardenerConfig.OidcConfig = &gqlschema.OIDCConfigInput{
+			ClientID:       r.oidcDefaultValues.ClientID,
+			GroupsClaim:    r.oidcDefaultValues.GroupsClaim,
+			IssuerURL:      r.oidcDefaultValues.IssuerURL,
+			SigningAlgs:    r.oidcDefaultValues.SigningAlgs,
+			UsernameClaim:  r.oidcDefaultValues.UsernameClaim,
+			UsernamePrefix: r.oidcDefaultValues.UsernamePrefix,
+		}
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/process/input/input.go:388]: also upgradeShootInput should be updated, let's do it when the conflict will be resolved in this PR: https://github.com/kyma-project/control-plane/pull/795

Comment (szwedm) [reply components/kyma-environment-broker/internal/process/input/input.go:388]: ok

**Review by piotrmiskiewicz — APPROVED** [review]

**Review by piotrmiskiewicz — APPROVED** [review]

**Review by piotrmiskiewicz — DISMISSED** [review]

File: docs/kyma-environment-broker/03-12-custom-oidc-configuration.md [hunk 2]
```diff
@@ -31,5 +31,99 @@ To create an SKR with a custom OIDC (Open ID Connect) configuration, specify the
        }
    }"
 ```
+>NOTE: `clientID` and `issuerURL` values are mandatory for custom OIDC configuration.
 
-If the `oidc` is not set, the default OIDC configuration is used.
+If you do not provide the `oidc` object in your provisioning request or leave all object's properties empty, the default OIDC configuration will be used.
+However, if you do the same in update request, it will leave the saved OIDC configuration untouched.
+In these particular cases, JSON could look like this:
```
Comment (mmitoraj) [inline docs/kyma-environment-broker/03-12-custom-oidc-configuration.md:38]:
```suggestion
See the following JSON example without the `oidc` object and with the `oidc` object whose properties are empty:
```

File: docs/kyma-environment-broker/03-12-custom-oidc-configuration.md [hunk 3]
```diff
@@ -31,5 +31,99 @@ To create an SKR with a custom OIDC (Open ID Connect) configuration, specify the
...
+      "issuerURL" : "",
+      "groupsClaim" : "",
+      "signingAlgs" : [],
+      "usernamePrefix" : "",
+      "usernameClaim" : ""
+    }
+  }
+}
+```
+The default OIDC configuration looks like this:
```
Comment (mmitoraj) [inline docs/kyma-environment-broker/03-12-custom-oidc-configuration.md:72]:
```suggestion
This is the default OIDC configuration:
```

File: docs/kyma-environment-broker/03-12-custom-oidc-configuration.md [hunk 4]
```diff
@@ -31,5 +31,99 @@ To create an SKR with a custom OIDC (Open ID Connect) configuration, specify the
...
+      "groupsClaim" : "groups",
+      "signingAlgs" : ["RS256"],
+      "usernamePrefix" : "-",
+      "usernameClaim" : "sub"
+    }
+  ...
+}
+```
+
+If you would like to update the OIDC configuration, please make sure that you provide values for the mandatory properties. Without these values, validation error will occur.
```
Comment (mmitoraj) [inline docs/kyma-environment-broker/03-12-custom-oidc-configuration.md:88]:
```suggestion
To update the OIDC configuration, provide values for the mandatory properties. Without these values, a validation error occurs.
```

File: docs/kyma-environment-broker/03-12-custom-oidc-configuration.md [hunk 5]
```diff
@@ -31,5 +31,99 @@ To create an SKR with a custom OIDC (Open ID Connect) configuration, specify the
...
+Update operation will overwrite the OIDC configuration values as provided in JSON. That means empty OIDC properties values are considered as valid values for configuration. Consider the following scenario:
```
Comment (mmitoraj) [inline docs/kyma-environment-broker/03-12-custom-oidc-configuration.md:89]:
```suggestion
The update operation overwrites the OIDC configuration values provided in JSON. It means that OIDC properties with empty values are considered valid. See the following scenario:
```

File: docs/kyma-environment-broker/03-12-custom-oidc-configuration.md [hunk 6]
```diff
@@ -31,5 +31,99 @@ To create an SKR with a custom OIDC (Open ID Connect) configuration, specify the
...
+
+   1. Existing instance has the following OIDC configuration:
```
Comment (mmitoraj) [inline docs/kyma-environment-broker/03-12-custom-oidc-configuration.md:91]:
```suggestion
   1. An existing instance has the following OIDC configuration:
```

File: docs/kyma-environment-broker/03-12-custom-oidc-configuration.md [hunk 7]
```diff
@@ -31,5 +31,99 @@ To create an SKR with a custom OIDC (Open ID Connect) configuration, specify the
...
+   ```
+   ClientID: 9bd05ed7-a930-44e6-8c79-e6defeb7dec9
+   IssuerURL: https://kymatest.accounts400.ondemand.com
+   GroupsClaim: groups
+   UsernameClaim: sub
+   UsernamePrefix: -
+   SigningAlgs: RS256
+   ```
+   2. User sends update request (HTTP PUT) with the following JSON in the payload:
```
Comment (mmitoraj) [inline docs/kyma-environment-broker/03-12-custom-oidc-configuration.md:100]:
```suggestion
   2. A user sends an update request (HTTP PUT) with the following JSON in the payload:
```

File: docs/kyma-environment-broker/03-12-custom-oidc-configuration.md [hunk 8]
```diff
@@ -31,5 +31,99 @@ To create an SKR with a custom OIDC (Open ID Connect) configuration, specify the
...
+      "issuerURL" : "https://new-issuer-url.local.com",
+      "groupsClaim" : "",
+      "signingAlgs" : [],
+      "usernamePrefix" : "",
+      "usernameClaim" : ""
+    }
+  }
+}
+```
+   3. The OIDC configuration is updated accordingly to the values of `oidc` object in JSON:
```
Comment (mmitoraj) [inline docs/kyma-environment-broker/03-12-custom-oidc-configuration.md:121]:
```suggestion
   3. The OIDC configuration is updated to include the values of the `oidc` object from JSON provided in the update request:
```

**Review by mmitoraj — CHANGES_REQUESTED** [review]
A left a few comments. Let me know if you have any questions. Thanks!

File: docs/kyma-environment-broker/03-12-custom-oidc-configuration.md [hunk 9]
```diff
@@ -31,5 +31,99 @@ To create an SKR with a custom OIDC (Open ID Connect) configuration, specify the
        }
    }"
 ```
+>NOTE: `clientID` and `issuerURL` values are mandatory for custom OIDC configuration.
 
-If the `oidc` is not set, the default OIDC configuration is used.
+If you do not provide the `oidc` object in your provisioning request or leave all object's properties empty, the default OIDC configuration will be used.
+However, if you do the same in update request, it will leave the saved OIDC configuration untouched.
```
[the remark this answers is not in the document]
Comment (szwedm) [reply docs/kyma-environment-broker/03-12-custom-oidc-configuration.md:37]: I meant if OIDC object is not provided or properties fields of OIDC object are left empty

[the remark this answers is not in the document]
Comment (mmitoraj) [reply docs/kyma-environment-broker/03-12-custom-oidc-configuration.md:37]: so lets put it that way: 
```suggestion
However, if you do not provide the `oidc` object in the update request or leave all object’s properties empty, the saved OIDC configuration stays untouched.
```

**Review by mmitoraj — APPROVED** [review]

File: docs/kyma-environment-broker/03-12-custom-oidc-configuration.md [hunk 10]
```diff
@@ -31,5 +31,99 @@ To create an SKR with a custom OIDC (Open ID Connect) configuration, specify the
        }
    }"
 ```
+>NOTE: `clientID` and `issuerURL` values are mandatory for custom OIDC configuration.
 
-If the `oidc` is not set, the default OIDC configuration is used.
+If you do not provide the `oidc` object in the provisioning request or leave all object's properties empty, the default OIDC configuration is used.
+However, if you do not provide the `oidc` object in the update request or leave all object’s properties empty, the saved OIDC configuration stays untouched.
+See the following JSON example without the `oidc` object and with the `oidc` object whose properties are empty:
```
Comment (mmitoraj) [inline docs/kyma-environment-broker/03-12-custom-oidc-configuration.md:38]:
```suggestion
See the following JSON example without the `oidc` object:
```

File: docs/kyma-environment-broker/03-12-custom-oidc-configuration.md [hunk 11]
```diff
@@ -31,5 +31,99 @@ To create an SKR with a custom OIDC (Open ID Connect) configuration, specify the
...
+  "plan_id" : "4deee563-e5ec-4731-b9b1-53b42d855f0c",
+  "context" : {
+    "globalaccount_id" : {GLOBAL_ACCOUNT_ID}
+  },
+  "parameters" : {
+    "name" : {CLUSTER_NAME}
+  }
+}
+```
+OR
```
Comment (mmitoraj) [inline docs/kyma-environment-broker/03-12-custom-oidc-configuration.md:51]:
```suggestion
See the following JSON example with the `oidc` object whose properties are empty:
```

**Review by mmitoraj — CHANGES_REQUESTED** [review]

**Review by mmitoraj — APPROVED** [review]

**Review by piotrmiskiewicz — APPROVED** [review]

**Pull request merged**